src/main/java/com/hierynomus/smbj/io/BufferedInputStreamReader.java

```diff
@@ -17,12 +17,14 @@
 
 import java.io.BufferedInputStream;
 import java.io.IOException;
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.hierynomus.smbj.common.SMBRuntimeException;
 
 public class BufferedInputStreamReader {
     private static final int EOF = -1;
 
-    private final AtomicBoolean isEnd = new AtomicBoolean(false);
+    private final AtomicInteger cachedByte = new AtomicInteger(EOF);
```

Comment:
```suggestion
    private volatile int cachedByte = EOF;
```

I don't think we can really make use of an atomic value here, a simple volatile variable should serve the same purpose, ready concurrently from different threads from a stream looks like waiting for ba things to happen, especially in combination with non atomic calls like aviable / read ...

Reply:
Yeah, volatile is enough. Actually it appears this object is not used across multiple threads so in theory we don't even need volatile, but better safe than sorry.